Source/WTF/Scripts/generate-platform-args

```diff
@@ -0,0 +1,91 @@
+    '{SYSTEM_HEADER_SEARCH_PATHS}'.format_map(os.environ))
+preprocessor_definitions = shlex.split(
+    'RELEASE_WITHOUT_OPTIMIZATIONS {GCC_PREPROCESSOR_DEFINITIONS}'.format_map(os.environ))
+archs = shlex.split(os.environ['ARCHS'])
+
+if os.environ['DEPLOYMENT_LOCATION'] == 'YES':
+    output_dir = '{DSTROOT}/{PRIVATE_HEADERS_FOLDER_PATH}/Scripts'.format_map(os.environ)
+else:
+    output_dir = '{BUILT_PRODUCTS_DIR}/{PRIVATE_HEADERS_FOLDER_PATH}/Scripts'.format_map(os.environ)
+
+combined_depfile_fd = open(
+    '{DERIVED_FILES_DIR}/generate-platform-args.d'.format_map(os.environ), 'w')
+
+# This script is run as a build phase, which Xcode runs once with
+# $arch=undefined, so we have to handle multi-arch builds manually. We generate
+# argument lists for all active architectures, and the build script phase
+# declares all known output paths as outputs.
+#
+# FIXME: As an alternative, we could integrate the script as a build rule in
+# targets that compile swift, and generate the arguments on demand.
+
+for arch in archs:
+    output_name = f'platform-enabled-swift-args.{arch}.resp'
+    output_file = f'{output_dir}/{output_name}'
+    depfile = f'{{DERIVED_FILES_DIR}}/{output_name}.d'.format_map(os.environ)
+    target_triple = (f'{arch}-'
+        '{LLVM_TARGET_TRIPLE_VENDOR}-{LLVM_TARGET_TRIPLE_OS_VERSION}').format_map(os.environ)
+    target_triple += os.environ.get('LLVM_TARGET_TRIPLE_SUFFIX', '')
+
+    feature_and_platform_defines = subprocess.check_output(
+        ('xcrun', 'clang', '-x', 'c++', '-std={CLANG_CXX_LANGUAGE_STANDARD}'.format_map(os.environ),
+         '-target', target_triple, '-E', '-P', '-dM', '-MD', '-MF', depfile, '-MT', output_name,
+         *(f'-D{token}' for token in preprocessor_definitions),
+         *(arg for path in framework_search_paths for arg in ('-F', path)),
+         *(arg for path in header_search_paths for arg in ('-I', path)),
+         *(arg for path in system_framework_search_paths for arg in ('-iframework', path)),
+         *(arg for path in system_header_search_paths for arg in ('-isystem', path)),
+         '-include', 'wtf/Platform.h', '/dev/zero'), text=True)
+
+    # Concatenate all the depfiles, in case a header is only imported on specific architectures.
+    shutil.copyfileobj(open(depfile), combined_depfile_fd)
+
+    definitions = {}
+    for m in re.finditer(r'^#define (\w+) (.+)', feature_and_platform_defines, re.MULTILINE):
+        name = m.group(1)
+        value = m.group(2)
+        if value == '1':
+            definitions[name] = 1
+        elif value == '0':
+            definitions[name] = 0
+        else:
+            definitions[name] = value
+
+    # Resolve macros which are defined as the value of another macro. clang -dM
+    # prints macros in alphabetical order, so we cannot rely on macros being in
+    # topological order. Limit the maximum amount of indirection to prevent a
+    # recursive definition from causing the script to loop forever.
```

Comment:
Should the script fail with an error when the maximum depth is reached?

Reply:
I don't think so, because there could be unrelated definitions coming out SDK headers that have no effect on platform defines, and we don't want those to tank the build.